### DISCUSSION TITLE
Pane tiling as shortcut problem

### DISCUSSION THREAD

**Discussion by gregorybleiker** [wezterm/wezterm#5006] [General]
I'm trying to split a window into 4 identical panes with a shortcut:
```
config.keys = {
  {
    key = 't',
    mods = 'CTRL',
    action = act.Multiple {
           wezterm.action.ToggleFullScreen,
           wezterm.action.SplitHorizontal { domain = 'CurrentPaneDomain' },
           act.ActivatePaneByIndex(0),
           wezterm.action.SplitVertical { domain = 'CurrentPaneDomain' },
           act.ActivatePaneByIndex(1),
           wezterm.action.SplitVertical { domain = 'CurrentPaneDomain' },
           },
  },
}
```
However, the `act.ActivatePaneByIndex` seem to be ignored. I didn't try the nightly build because this seems to by "by design"?

**Comment by wez** [comment]
certain key assignment actions are asynchronous, but when executed in the context of dispatching a key assignment, they are not (because they cannot be, in that context) awaited.

If you want to script a more complex sequence of events like this I would recommend using `wezterm.action_callback` to execute a lua function, and from there to use the mux API functions instead of the key assignments.

https://github.com/wez/wezterm/discussions/2388#discussioncomment-3371519  is a relevant/related discussion with a brief example that should get you started.

**Reply by gregorybleiker** [reply]
That looks exactly right, @wez, thanks a lot!

**Reply by gregorybleiker** [reply]
This is what I ended up with
```
config.keys = {
  {
    key = 't',
    mods = 'CTRL',
    action = act.Multiple {
           wezterm.action_callback(function(win, pane)
            local new_pane = pane:split{ direction = 'Right' }
            pane:split {direction = 'Top'}
            new_pane:split {direction = 'Top'}
            end),
           act.ActivatePaneByIndex(1),
           },
  },
}
```

**Reply by gregorybleiker** [reply]
Actually, the `Multiple` is not needed, so
```
config.keys = {
  {
    key = 't',
    mods = 'CTRL',
    action = wezterm.action_callback(function(win, pane)
            local new_pane = pane:split{ direction = 'Right' }
            pane:split {direction = 'Top'}
            new_pane:split {direction = 'Top'}
            end)
  },
}
```
